INT-B-22133 prime simulator update port

brianmanley-caci opened the pull request:
## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1047000)

## Summary
This story adds the functionality for updating the Port on a service item (PODFSC or POEFSC) in the prime simulator.

Is there anything you would like reviewers to give additional scrutiny?

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Create an HHG shipment with Alaska as either the origin or destination.
2. Copy down the move code.
3. Approved the shipment as a TOO.
4. Login to the office app as a Prime Simulator user.
5. Search for the Move code from step 2.
6. Click on the row with the move.
7. Scroll down to the shipment you created in step 1, and locate the International POD (or POE) Fuel Surcharge.
8. Click the Edit button beside the service item.
9. Verify the Port, Port Location, and Port Code fields are all empty.
10. Enter a valid port code (ports.port_code).
11. Click Save.
12. Verify you receive a success message: "Successfully updated service item".
13. Scroll back down to the service item and click the edit button again.
14. You should now see the Port, Port Location and Port Code populated.
15. Try saving with an invalid port code and you should receive an error message.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

### Database

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
![image](https://github.com/user-attachments/assets/7c79375b-eba2-4913-ae10-0ecc3c6d79a0)

![image](https://github.com/user-attachments/assets/e907bd87-bd21-4f50-ad65-3b9c9380a761)

![image](https://github.com/user-attachments/assets/13dcd70e-f090-487d-aabc-d48f71b99e63)

![image](https://github.com/user-attachments/assets/cef2c366-404f-47ce-b5a2-abc6275087fc)

danieljordan-caci reviewed (changes requested):
Could you handle lowercase values? `pdx` and `sea` should work imo.

![Screenshot 2025-01-03 at 2 30 19 PM](https://github.com/user-attachments/assets/94087585-6271-4817-ae5b-e602c7f85e0a)

![Screenshot 2025-01-03 at 2 30 09 PM](https://github.com/user-attachments/assets/c724a2be-51e0-4b69-9a70-a13c4e687a83)

brianmanley-caci commented:
> Could you handle lowercase values? `pdx` and `sea` should work imo.
> 
Since this prime sim tool exists to make our lives easier, I would agree with that idea.

msaki-caci commented:
2 things I noticed both are probably insignificant.

1. In the BL it states that the edit page for the POE/POD FSC should be the "**_Update Port Service Item page"_** But in your code it is the **_"Update International Fuel Surchage Service Item"_**.

2. Also in the BL it states on the edit page we should see:
"ID (readonly)
MTO ID (readonly)
Shipment ID (readonly)
Status (readonly)
Port (readonly)
**Port (readonly)**
Port Code"

I am assuming that it should be Port Location. 

In either case you just might want to update the BL. Otherwise everything looks good to me.

brianmanley-caci commented:
> > Could you handle lowercase values? `pdx` and `sea` should work imo.
> 
> Since this prime sim tool exists to make our lives easier, I would agree with that idea.

Added logic to convert to uppercase, and added a test for that.

brianmanley-caci commented:
> 2 things I noticed both are probably insignificant.
> 
> 1. In the BL it states that the edit page for the POE/POD FSC should be the "**_Update Port Service Item page"_** But in your code it is the **_"Update International Fuel Surchage Service Item"_**.
> 2. Also in the BL it states on the edit page we should see:
>    "ID (readonly)
>    MTO ID (readonly)
>    Shipment ID (readonly)
>    Status (readonly)
>    Port (readonly)
>    **Port (readonly)**
>    Port Code"
> 
> I am assuming that it should be Port Location.
> 
> In either case you just might want to update the BL. Otherwise everything looks good to me.

Thanks for catching that. I went back and fixed both of those on the story.

danieljordan-caci reviewed (approved):
Looks good! Much better - thanks for the fixes.
